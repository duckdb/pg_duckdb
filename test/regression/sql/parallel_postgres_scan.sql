CREATE TABLE tbl (id int, c float8, d text);
INSERT INTO tbl SELECT i, i * 2.0, 'helloworld' FROM generate_series(1, 1000000) i;
SELECT * FROM tbl ORDER BY 1,2,3 LIMIT 3;
DROP TABLE tbl;

-- JSON/LIST type
CREATE TABLE tbl (id int, c jsonb, d text[]);
INSERT INTO tbl SELECT i, jsonb_build_object('a', i), array_agg(i) FROM generate_series(1, 500000) i GROUP BY i;
SELECT * FROM tbl ORDER BY id DESC LIMIT 3;

CREATE TABLE tbl1 (id int, c jsonb, d text[]);
INSERT INTO tbl1 SELECT i, jsonb_build_object('a', i), array_agg(i) FROM generate_series(1, 100000) i GROUP BY i;
SELECT tbl.id, tbl.c, tbl.d, tbl1.c, tbl1.d FROM tbl JOIN tbl1 ON tbl.id = tbl1.id ORDER BY 1,2,3,4 LIMIT 5;

DROP TABLE tbl, tbl1;

