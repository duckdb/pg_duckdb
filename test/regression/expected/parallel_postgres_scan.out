CREATE TABLE tbl (id int, c float8, d text);
INSERT INTO tbl SELECT i, i * 2.0, 'helloworld' FROM generate_series(1, 1000000) i;
SELECT * FROM tbl ORDER BY 1,2,3 LIMIT 3;
 id | c |     d      
----+---+------------
  1 | 2 | helloworld
  2 | 4 | helloworld
  3 | 6 | helloworld
(3 rows)

DROP TABLE tbl;
-- JSON/LIST type
CREATE TABLE tbl (id int, c jsonb, d text[]);
INSERT INTO tbl SELECT i, jsonb_build_object('a', i), array_agg(i) FROM generate_series(1, 500000) i GROUP BY i;
SELECT * FROM tbl ORDER BY id DESC LIMIT 3;
   id   |       c       |    d     
--------+---------------+----------
 500000 | {"a": 500000} | {500000}
 499999 | {"a": 499999} | {499999}
 499998 | {"a": 499998} | {499998}
(3 rows)

CREATE TABLE tbl1 (id int, c jsonb, d text[]);
INSERT INTO tbl1 SELECT i, jsonb_build_object('a', i), array_agg(i) FROM generate_series(1, 100000) i GROUP BY i;
SELECT tbl.id, tbl.c, tbl.d, tbl1.c, tbl1.d FROM tbl JOIN tbl1 ON tbl.id = tbl1.id ORDER BY 1,2,3,4 LIMIT 5;
 id |    c     |  d  |    c     |  d  
----+----------+-----+----------+-----
  1 | {"a": 1} | {1} | {"a": 1} | {1}
  2 | {"a": 2} | {2} | {"a": 2} | {2}
  3 | {"a": 3} | {3} | {"a": 3} | {3}
  4 | {"a": 4} | {4} | {"a": 4} | {4}
  5 | {"a": 5} | {5} | {"a": 5} | {5}
(5 rows)

DROP TABLE tbl, tbl1;
